types/jasmine/index.d.ts

```diff
@@ -948,92 +838,41 @@ declare namespace jasmine {
         jasmineDone?(runDetails: RunDetails): void;
     }
 
-    interface Runner {
-        new (env: Env): any;
-
-        execute(): void;
-        beforeEach(beforeEachFunction: SpecFunction): void;
-        afterEach(afterEachFunction: SpecFunction): void;
-        beforeAll(beforeAllFunction: SpecFunction): void;
-        afterAll(afterAllFunction: SpecFunction): void;
-        finishCallback(): void;
-        addSuite(suite: Suite): void;
-        add(block: Block): void;
-        specs(): Spec[];
-        suites(): Suite[];
-        topLevelSuites(): Suite[];
-        results(): NestedResults;
-    }
-
     type SpecFunction = (spec?: Spec) => void;
 
     interface SuiteOrSpec {
         id: number;
         env: Env;
         description: string;
-        queue: Queue;
     }
 
     interface Spec extends SuiteOrSpec {
-        new (env: Env, suite: Suite, description: string): any;
-
-        suite: Suite;
-
-        afterCallbacks: SpecFunction[];
-        spies_: Spy[];
-
-        results_: NestedResults;
-        matchersClass: Matchers<any>;
-
         getFullName(): string;
-        results(): NestedResults;
-        log(arguments: any): any;
-        runs(func: SpecFunction): Spec;
-        addToQueue(block: Block): void;
-        addMatcherResult(result: Result): void;
         getResult(): any;
         expect(actual: any): any;
-        waits(timeout: number): Spec;
-        waitsFor(latchFunction: SpecFunction, timeoutMessage?: string, timeout?: number): Spec;
-        fail(e?: any): void;
-        getMatchersClass_(): Matchers<any>;
-        addMatchers(matchersPrototype: CustomMatcherFactories): void;
-        finishCallback(): void;
-        finish(onComplete?: () => void): void;
-        after(doAfter: SpecFunction): void;
         execute(onComplete?: () => void, enabled?: boolean): any;
-        addBeforesAndAftersToQueue(): void;
-        explodes(): void;
-        spyOn(obj: any, methodName: string, ignoreMethodDoesntExist: boolean): Spy;
-        spyOnProperty(object: any, property: string, accessType?: "get" | "set"): Spy;
-        spyOnAllFunctions(object: any): Spy;
-
-        removeAllSpies(): void;
         throwOnExpectationFailure: boolean;
     }
 
     interface XSpec {
         id: number;
-        runs(): void;
     }
 
+    /**
+     * @deprecated Private. Use the global `describe`, `beforeEach`, `it`, etc functions instead.
```

Comment:
I don't know the best practices but I don't think this should be deprecated unless it is going to change soon. When using jasmine it is not very useful to get the Suite but it is very useful when creating a reporter.

Reply:
Can you point me at an example? The thing that Jasmine passes to reporters [isn't actually a suite](https://jasmine.github.io/api/edge/global.html#SuiteResult), but maybe I'm not thinking of the same thing you are.

Reply:
[atom-jasmine3-test-runner](https://github.com/UziTech/atom-jasmine3-test-runner) needs to use the suite to display the suite description. We get the suites and specs from `jasmine.getEnv().topSuite()`.

![image](https://user-images.githubusercontent.com/97994/113485849-4b479900-9475-11eb-82b7-12061e4a4275.png)

Reply:
Ok, that helps. It looks like `atom-jasmine3-test-runner` is building up UI based on the structure of the suites and specs before the first spec starts. That can't be done using just the reporter interface. I'll remove the deprecation. Longer term, I think the right move is to add the parts of Suite that are useful for this sort of thing to the Jasmine API docs, and mark the other members deprecated here.

Reply:
>I don't know the best practices but I don't think this should be deprecated unless it is going to change soon. 

This obviously would show in IDEs like VSCode as deprecated, which could trick users. I've proposed to consider this, if you're to remove that from the type definition itself for methods, which are identified as wrong. The `deprecated` can be not the best choice for this reason, though. thx!

Reply:
@UziTech I took a quick pass at the doc additions. Let me know if you think anything else should be added: https://jasmine.github.io/api/edge/Suite.html

Reply:
The docs look good to me.